File: Surface/Surface/SfBox.m
//
//  SfBox.m
//  Surface
//
//  Created by Webchimp on 22/03/16.
//  Copyright © 2016 Webchimp. All rights reserved.
//

#import "SfBox.h"

@implementation SfBox


- (id) init
{
    self = [super init];
    if (self) {
        
        self.top = 0;
        self.right = 0;
        self.bottom = 0;
        self.left = 0;
        
    }
    return self;
}

- (void)setBox:(float)top Right:(float)right Bottom:(float)bottom Left:(float)left {
    
    self.top = top;
    self.right = right;
    self.bottom = bottom;
    self.left = left;
}

- (float)getWidth {
    
    return self.right + self.left;
}

- (float)getHeight {
    
    return self.top + self.bottom;
}

- (BOOL)isEmpty {
    
    if ( !self.top && !self.right && !self.bottom && !self.left ) {
        return true;
    }
    return false;
}

@end

File: Swift/SurfaceEngineDemo/SurfaceEngineDemo/SfBox.swift
//
//  SfBox.swift
//  SurfaceEngine
//
//  Created by Erick Sanchez on 11/10/17.
//  Copyright © 2017 App Builders. All rights reserved.
//

import Foundation

public class SfBox {
    
    var top: Float!;
    var right: Float!;
    var bottom: Float!;
    var left: Float!;
    
    init() {
        
        self.top = 0.0;
        self.right = 0.0;
        self.bottom = 0.0;
        self.left = 0.0;
    }
    
    convenience init(top:Float, right:Float, bottom:Float, left:Float) {
        
        self.init();
        self.setBox(top: top, right: right, bottom: bottom, left: left);
    }
    
    func setBox(top:Float, right:Float, bottom:Float, left:Float) -> Void {
        
        self.top = top;
        self.right = right;
        self.bottom = bottom;
        self.left = left;
    }
    
    func getWidth() -> Float {
        
        return self.left + self.right;
    }
    
    func getHeight() -> Float {
        
        return self.top + self.bottom;
    }
    
    func isEmpty() -> Bool {
        
        return (self.top == 0 && self.right == 0 && self.bottom == 0 && self.left == 0);
    }
}
